scripts/deploy.sh

```diff
@@ -1,5 +1,18 @@
 # This script runs the aws cli command to deploy the app to s3
 # It also uploads source maps to bugsnag
+BASE_URL=$REACT_APP_PARSLEY_URL
+# First download the currently deployed commit hash from s3
+# If the curl fails, then use the previous tag from git
+# If the curl succeeds, then use the githash from the curl
+PREVIOUS_DEPLOYED_COMMIT=$(curl -s $BASE_URL/commit.txt)
+
+# Save the current commit hash to a file
+echo "Saving current commit hash to file"
+echo $PREVIOUS_DEPLOYED_COMMIT > bin/previous_deploy.txt
+
+# Create a file in dist/ that contains the current commit hash
+CURRENT_COMMIT_HASH=$(git rev-parse HEAD)
+echo $CURRENT_COMMIT_HASH > dist/commit.txt
```

Comment:
Writing a new file feels like a roundabout way of saving this information. Did you consider [a `<meta>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta)? Something like this in `index.html` could work:

```html
<meta name="version" content={REACT_APP_VERSION} />
```

There may be a more significant lift with fetching and parsing this, though. Interested in your thoughts!

Reply:
[We actually already do this](https://github.com/evergreen-ci/parsley/blob/05feeb479651bdab46342814aafd615f5ebd48ab/index.html#L7)!
 I considered using the meta tag but the lift seemed a bit too complex for a shell script. I would have to curl the HTML file and then figure out a way to parse that tag. The _simplest_ option is [to use a regex to parse the HTML](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags) but I don't think that would be as good of an idea as just reading a plain text file.

Do you think its something that would be worth trying?

Reply:
Yeah the regex does feel messier overall. This approach works for me!